haystack/components/agents/agent.py

```diff
@@ -187,6 +188,33 @@ def from_dict(cls, data: Dict[str, Any]) -> "Agent":
 
         return default_from_dict(cls, data)
 
+    def _prepare_generator_inputs(self, streaming_callback: Optional[StreamingCallbackT] = None) -> Dict[str, Any]:
+        """Prepare inputs for the chat generator."""
+        generator_inputs = {"tools": self.tools}
+        selected_callback = streaming_callback or self.streaming_callback
+        if selected_callback is not None:
+            generator_inputs["streaming_callback"] = selected_callback
+        return generator_inputs
+
+    def _create_agent_span(self, input_data: Dict[str, Any]) -> Iterator[tracing.Span]:
+        """Create a span for the agent run."""
+        return tracing.tracer.trace(
+            "haystack.agent.run",
+            tags={
+                "haystack.agent.input_data": input_data,
+                "haystack.agent.max_steps": self.max_agent_steps,
+                "haystack.agent.tools": self.tools,
+                "haystack.agent.exit_conditions": self.exit_conditions,
+                "haystack.agent.state_schema": self.state_schema,
+            },
+        )
+
+    def _finalize_run(self, state: State, span: tracing.Span, counter: int) -> Dict[str, Any]:
+        """Finalize the agent run and return the state data."""
+        span.set_content_tag("haystack.agent.output_data", state.data)
+        span.set_tag("haystack.agent.steps_taken", counter)
+        return state.data
```

Comment:
Perhaps it does make sense to do this, but it hides and adds layer of indirection for a reader trying to understand how Agent works

Reply:
Fair enough I’d also be happy just getting rid of this util function.